tflearn/layers/recurrent.py

```diff
@@ -514,7 +515,13 @@ def __call__(self, inputs, state, scope=None):
             new_c = (c * self._inner_activation(f + self._forget_bias) +
                      self._inner_activation(i) *
                      self._activation(j))
-            new_h = self._activation(new_c) * self._inner_activation(o)
+            
+            # hidden-to-hidden batch normalizaiton
+            if batch_norm == False:
+                new_h = self._activation(new_c) * self._inner_activation(o)
+            else:
+                batch_norm_new_c = batch_normalization(new_c, restore=False)
+                new_h = self._activation(batch_norm_new_c) * self._inner_activation(o)
```

Comment:
This changes the meaning though, since `new_c` is a return value. Previously the returned cell value was not normalized (as cf the paper it shouldn't be), now it is.